src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs

```diff
@@ -278,6 +280,17 @@ private IPlaceholderExpression CreateBoundDeconstructValuePlaceholderOperation(B
             return new PlaceholderExpression(_semanticModel, syntax, type, constantValue, isImplicit);
         }
 
+        private IDeconstructionAssignmentExpression CreateBoundDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator boundDeconstructionAssignmentOperator)
+        {
+            Lazy<IOperation> target = new Lazy<IOperation>(() => Create(boundDeconstructionAssignmentOperator.Left));
+            Lazy<IOperation> value = new Lazy<IOperation>(() => Create(boundDeconstructionAssignmentOperator.Right));
```

Comment:
>boundDeconstructionAssignmentOperator.Right [](start = 71, length = 43)

It feels like we should dig through this conversion because it is always synthetic Deconstruction conversion.

Reply:
Makes sense. In future, we can expose this as a conversion or something else based on how deconstruction is represented in the language spec. I will make this change.